EuclideanGeometry/Foundation/Axiom/Linear/Perpendicular.lean

```diff
@@ -101,4 +101,31 @@ theorem dist_pt_line_shortest (A B : P) {l : Line P} (h : B LiesOn l) : dist A B
 
 end Perpendicular_constructions
 
+section Perpendicular_inner_product
+
+theorem perp_of_inner_product_eq_zero (v w : VecND) (h : inner v.1 w.1 = (0 : ℝ)) : v ⟂ w := by
```

Comment:
This theorem can be enhanced to an iff version.

Reply:
I will prove the reverse implication in a separate commit.